### PR TITLE
feat(to_unixtime): add initial implementation

### PR DESCRIPTION
**Pull request by etolbakov**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

The feature adds a new UDF `to_unixtime()` for conversions from a timestamp in RFC3339 format to a Unix timestamp.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
https://github.com/GreptimeTeam/greptimedb/issues/1103

### CODE REVIEW

File: src/common/function/src/scalars/timestamp.rs [hunk 1]
```diff
@@ -11,11 +11,18 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+use std::sync::Arc;
+mod from_unixtime;
```
Comment (etolbakov) [inline src/common/function/src/scalars/timestamp.rs:15]: will remove it in the next push

File: src/common/function/src/scalars/timestamp.rs [hunk 2]
```diff
@@ -11,11 +11,18 @@
...
+
+use to_unixtime::ToUnixtimeFunction;
 
 use crate::scalars::function_registry::FunctionRegistry;
 
 pub(crate) struct TimestampFunction;
 
 impl TimestampFunction {
-    pub fn register(_registry: &FunctionRegistry) {}
+    pub fn register(_registry: &FunctionRegistry) {
```
Comment (etolbakov) [inline src/common/function/src/scalars/timestamp.rs:25]: fix it in the next push

**Comment by etolbakov** [comment]
@waynexia 
Could you please take a look when you have time?

File: src/common/function/src/scalars/timestamp/to_unixtime.rs [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+const NAME: &str = "to_unixtime";
+
+fn convert_to_seconds(arg: &str) -> Option<i64> {
+    match Timestamp::from_str(arg) {
+        Ok(ts) => {
+            let sec_mul = (TimeUnit::Second.factor() / ts.unit().factor()) as i64;
+            Some(ts.value().div_euclid(sec_mul))
+        }
+        Err(_err) => {
+            // error!("Failed to parse {} to Timestamp value", arg);
```
Comment (etolbakov) [inline src/common/function/src/scalars/timestamp/to_unixtime.rs:41]: not too sure if it worth logging, happy to remove/adjust it

File: src/common/function/src/scalars/timestamp/to_unixtime.rs [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+    fn signature(&self) -> Signature {
+        Signature::exact(
+            vec![ConcreteDataType::String(StringType)],
+            Volatility::Immutable,
+        )
+    }
+
+    fn eval(&self, _func_ctx: FunctionContext, columns: &[VectorRef]) -> Result<VectorRef> {
+        match columns[0].data_type() {
```
Comment (waynexia) [inline src/common/function/src/scalars/timestamp/to_unixtime.rs:64]: We can verify the input `columns`' length, and report an `InvalidFuncArgs` error if there is more than one column.

File: src/common/function/src/scalars/timestamp/to_unixtime.rs (hunk 3 above)
Comment (waynexia) [reply src/common/function/src/scalars/timestamp/to_unixtime.rs:41]: IMO it's acceptable to disregard this conversion error. They will disappear in the result column which may indicate that something is not working correctly.

File: src/common/function/src/scalars/timestamp/to_unixtime.rs [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    }
+}
+
+impl Function for ToUnixtimeFunction {
+    fn name(&self) -> &str {
+        NAME
+    }
+
+    fn return_type(&self, _input_types: &[ConcreteDataType]) -> Result<ConcreteDataType> {
+        Ok(ConcreteDataType::int64_datatype())
```
Comment (waynexia) [inline src/common/function/src/scalars/timestamp/to_unixtime.rs:53]: For clearness I'd prefer to return `ConcreteDataType::timestamp_second_datatype()` data type. `Int64` defaults to millisecond resolution somewhere (🥲

File: src/common/function/src/scalars/timestamp/to_unixtime.rs [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+                            volatility: Volatility::Immutable
+                         } if  valid_types == vec![ConcreteDataType::String(StringType)]
+        ));
+
+        let times = vec![
+            Some("2023-03-01T06:35:02Z"),
+            None,
+            Some("2022-06-30T23:59:60Z"),
+            Some("invalid_time_stamp"),
+        ];
```
Comment (waynexia) [inline src/common/function/src/scalars/timestamp/to_unixtime.rs:121]: cooool 👍

**Review by waynexia — COMMENTED** [review]
Great job! This patch generally looks good to me. Do you mind also adding some cases in our [sqlness](https://github.com/GreptimeTeam/greptimedb/blob/develop/tests/README.md) suite? Maybe a new file under [common/select](https://github.com/GreptimeTeam/greptimedb/tree/develop/tests/cases/standalone/common/select) dir.

**Review by evenyag — APPROVED** [review]
LGTM

**Comment by etolbakov** [comment]
@waynexia 
Could you please take a look at the changes I've made based on your feedback?

File: tests/cases/standalone/common/select/dummy.sql [hunk 7]
```diff
@@ -9,3 +9,5 @@ select "a";
 select "A";
 
 select * where "a" = "A";
+
+select TO_UNIXTIME('2023-03-01T06:35:02Z');
```
Comment (waynexia) [inline tests/cases/standalone/common/select/dummy.sql:13]:
```suggestion
select TO_UNIXTIME('2023-03-01T06:35:02Z');

```

File: tests/README.md [hunk 8]
```diff
@@ -3,7 +3,7 @@
 ## Sqlness manual
 
 ### Case file
-Sqlness has three types of file
+Sqlness has two types of file
```
Comment (waynexia) [inline tests/README.md:6]: nice catch 🤣

**Review by waynexia — APPROVED** [review]
Thanks for your great work ❤️

File: tests/README.md (hunk 8 above)
Comment (etolbakov) [reply tests/README.md:6]: I took it as a "test" to see I check the docs before doing stuff 😅

**Pull request merged**